pkgs/development/ruby-modules/bundler-env/default.nix

```diff
@@ -67,25 +76,56 @@ let
     if hasBundler then gems.bundler
     else defs.bundler.override (attrs: { inherit ruby; });
 
-  gems = lib.flip lib.mapAttrs configuredGemset (name: attrs:
-    buildRubyGem ((removeAttrs attrs ["source"]) // attrs.source // {
-      inherit ruby;
-      gemName = name;
-      gemPath = map (gemName: gems."${gemName}") (attrs.dependencies or []);
-    }));
+  pathDerivation = {
+    usesGemspec ? false, ...
+  }@attrs:
+    let
+      res = {
+          inherit usesGemspec;
+          type = "derivation";
+          name = attrs.gemName;
+          version = attrs.version;
+          outPath = attrs.path;
+          outputs = [ "out" ];
+          out = res;
+          outputName = "out";
+        };
+    in res;
+
+  buildGem = name: attrs: (
+    let
+      gemAttrs = ((removeAttrs attrs ["source"]) // attrs.source // {
+        inherit ruby;
+        gemName = name;
+        gemPath = map (gemName: gems."${gemName}") (attrs.dependencies or []);
+      });
+    in
+    if gemAttrs.type == "path" then pathDerivation gemAttrs
+    else buildRubyGem gemAttrs);
+
+  gems = lib.flip lib.mapAttrs configuredGemset (name: attrs: buildGem name attrs);
+
+  maybeCopyAll = main: if main == null then "" else copyIfUseGemspec main;
+
+  copyIfUseGemspec = { usesGemspec ? false, ...}@main:
```

Comment:
Of course, if I'd read with a little more context...

Yes, any mainGem that has a path source is going to need everything copied in. Furthermore, because of how Bundler handles `gemspec` in the first place, the two kinds of gems are indistinguishable (Bundler parses a Gemfile/lock with `gemspec` as if it were `gem "name", :path => .`, effectively.)